librosa/core/constantq.py

```diff
@@ -552,3 +551,19 @@ def __early_downsample(y, sr, hop_length, res_type, n_octaves,
         sr = sr // downsample_factor
```

Comment:
Good point; I suppose there's no reason to force sr to be integer-typed.

There may be points elsewhere in the library that check for this though.  I'll investigate and maybe start a separate issue for that.